run.sh

```diff
@@ -37,15 +37,25 @@ function output() {
 }
 
 oneShot=${oneShot:=n}
+# empty (disabled), irqs, scheduling, or full (see hook.sh)
+ftraceMode=${ftraceMode:=""}
+maxFtraces=${maxFtraces:=0}
 intervalSecs=${intervalSecs:=60}
 journalSocket=${journalSocket:=/run/systemd/journal/stdout}
 hostdev=${hostdev:=/hostdev}
 console=${hostdev}/console
 logs=${logs:=/logs}
+
 if [[ -z ${bytesMax} ]]; then
   bytesMax=$(( 250 * 1000 * 1000 )) # Maximum size of logs to keep around.
 fi
 
+if [[ -n "$ftraceMode" ]]; then
+  nsenter -t 1 -m -u -i -n -p apt-get update
+  nsenter -t 1 -m -u -i -n -p apt-get install -y trace-cmd
```

Comment:
what is this doing? can we apt-get in the Dockerfile instead?

Reply:
I think ftrace needs to run within the root namespace, so this will perform the apt-get update and install on the actual host by entered root namespaces (for pid 1).

Reply:
gotcha - that makes sense (although i don't like the idea of installing stuff on the host from the container).

Reply:
Can we build this into the image or there are kernel modules etc?